libraries/tentacle.rb

```diff
@@ -41,10 +43,31 @@ def installer_url(version)
       "https://download.octopusdeploy.com/octopus/Octopus.Tentacle.#{version}-x64.msi"
     end
 
+    def tentacle_exists?(server, api_key, thumbprint)
+      ::JSON.parse(api_client(server, api_key).get('/machines/all')).any? do |machine|
```

Comment:
/machines/all can take a long time to return. I don't know how well that will scale to hundreds of tentacles. 

I've opened a feature request to allow for querying a single tentacle instance to see if it is already registered. 

http://help.octopusdeploy.com/discussions/beta-testing-feedback/167-feature-request-api-to-querry-if-tentacle-already-exists